### PR TITLE
Add serde feature

### PR DESCRIPTION
**Pull request by d-sonuga**
For dusk-network/rusk/issues/2773 and https://github.com/dusk-network/jubjub/issues/143.

~~To serialize `AffineNielsPoint` and `ExtendedNielsPoint`, I implemented `dusk_bytes::Serializable` for them.
But there is currently some uncertainty about the correctness of the approach used to serialize because I don't understand `AffinePoint`'s serialization and I'm unsure about whether or not it will be relevant to `AffineNielsPoint` and `ExtendedNielsPoint`.~~
Edit: `AffineNielsPoint` and `ExtendedNielsPoint` have not being changed.

### CODE REVIEW

**Review by Neotamandua — COMMENTED** [review]
LGTM from my side, but I'm waiting for a second approval from a maintainer of this repository

File: src/dusk/serde_support.rs
Comment (Neotamandua) [inline src/dusk/serde_support.rs:54]:
```suggestion
        AffinePoint::deserialize(deserializer).map(Into::into)
```

one small nit though, this can also be done and avoids creating an unnecessary closure.

File: src/dusk/serde_support.rs
Comment (moCello) [inline src/dusk/serde_support.rs:82]: Please add all additional implementations on `Fr` under `fr::dusk`

File: .github/workflows/dusk_ci.yml
Comment (moCello) [inline .github/workflows/dusk_ci.yml:34]: I would not add another test run but add the `serde` feature to the `test_nightly_std` tests

**Review by Neotamandua — COMMENTED** [review]
Generally, looks good to me. But based on the other two PRs (https://github.com/dusk-network/bls12_381-bls/pull/22 & https://github.com/dusk-network/jubjub-schnorr/pull/30), and for consistency, I would suggest having the tests as integration tests rather than module tests. Also because the current module tests are testing the implementation for `Fr`, which is in another module.

Edit: After checking back, we can leave it as it is except for the tests for `Fr`. Can you move them into the respective Dusk.rs file of the Fr module?

File: src/dusk/serde_support.rs
Comment (moCello) [inline src/dusk/serde_support.rs:90]: The tests for `Fr` also needs to be in `fr::dusk`.
I wouldn't mind if you add them to the `serde_fr` function though if you add a comment on why they are expected to fail.

File: src/dusk/serde_support.rs
Comment (d-sonuga) [reply src/dusk/serde_support.rs:87]: Yeah, but what if its implementation is changed later?

**Review by moCello — COMMENTED** [review]
Code looks good to me, but please also update the issue and pr description and add an entry to the changelog

File: src/dusk/serde_support.rs
Comment (moCello) [inline src/dusk/serde_support.rs:65]: we also need a test for a non-identity point. In your previous version this was done with an extended point

File: src/dusk/serde_support.rs
Comment (Neotamandua) [inline src/dusk/serde_support.rs:64]: It is better imo to use [SeedableRng](https://rust-random.github.io/rand/rand_core/trait.SeedableRng.html) here as well like in other tests you did.

**Review by Neotamandua — CHANGES_REQUESTED** [review]

File: src/fr/dusk.rs
Comment (Neotamandua) [inline src/fr/dusk.rs:414]: same here

File: src/dusk/serde_support.rs
Comment (Neotamandua) [inline src/dusk/serde_support.rs:28]: Can't we use Self::SIZE here as well instead of relying on &"32"?

File: src/fr/dusk.rs
Comment (Neotamandua) [inline src/fr/dusk.rs:308]: same here

**Review by Neotamandua — APPROVED** [review]
LGTM.

We can now aim for a 0.14.2 release of this crate tomorrow, so that we can proceed directly with the phoenix-core crate that depends on it.

**Pull request merged**
